surface/include/pcl/surface/advancing_front_utils.h

```diff
@@ -0,0 +1,310 @@
+    {
+      return pcl::PointXYZ (p (0), p (1), p (2));
+    }
+
+    /** \brief Convert Eigen Vector3f to PCL PointXYZ */
+    pcl::PointXYZ
+    convertEigenToPCL (const Eigen::Vector3d &p)
+    {
+      return pcl::PointXYZ (p (0), p (1), p (2));
+    }
+
+    AfrontVertexPointType
+    convertPointNormalToAfrontPointType (const pcl::PointNormal &p)
+    {
+      AfrontVertexPointType result;
+      result.x = p.x;
+      result.y = p.y;
+      result.z = p.z;
+      result.data[3] = 1.0f;
+      result.normal_x = p.normal_x;
+      result.normal_y = p.normal_y;
+      result.normal_z = p.normal_z;
+      result.data_n[3] = 0.0f;
+      result.curvature = p.curvature;
+      result.max_step = 0.0f;
+      result.max_step_search_radius = 0.0f;
+      return result;
+    }
+
+    AfrontGuidanceFieldPointType
+    convertAfrontPointTypeToAfrontGuidanceFieldPointType (const AfrontVertexPointType &p, const double rho)
+    {
+      pcl::afront::AfrontGuidanceFieldPointType result;
+      result.x = p.x;
+      result.y = p.y;
+      result.z = p.z;
+      result.data[3] = 1.0f;
+      result.normal_x = p.normal_x;
+      result.normal_y = p.normal_y;
+      result.normal_z = p.normal_z;
+      result.data_n[3] = 0.0f;
+      result.curvature = p.curvature;
+      result.ideal_edge_length = 2.0 * std::sin (rho / 2.0) / result.curvature;
+      return result;
+    }
+
+    /**
+      * \brief Get the mid point of a half edge given it's verticies
+      * \param[in] p1 Vertex of half edge
+      * \param[in] p2 Vectex of half edge
+      * \return The mid point of the half edge
+      */
+    Eigen::Vector3f
+    getMidPoint (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2)
+    {
+      return (p1 + p2) / 2.0;
+    }
+
+    /**
+      * \brief Get the distance between two points
+      * \param[in] p1 Vertex of half edge
+      * \param[in] p2 Vectex of half edge
+      * \return The lenght of the half edge
+      */
+    double
+    distPoint2Point (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2)
+    {
+      return (p2 - p1).norm ();
+    }
+
+    struct DistPoint2LineResults
+    {
+      Eigen::Vector3f points[3]; /**< \brief The points. Note: Seg = (points[1] - points[0]) */
+      Eigen::Vector3f p;         /**< \brief The closest point on line segment. */
+      float mu;                  /**< \brief The corresponding line paramentric value resulting in the minimum distance. */
+      double d;                  /**< \brief The distance between point and line segments */
+    };
+
+    /**
+      * \brief Get the distance between a point and line
+      * \param[in] p1 Origin point of the line
+      * \param[in] p2 Terminating point of the line
+      * \param[in] p The point for calulating the distance to the line
+      * \return DistPoint2LineResults
+      */
+    DistPoint2LineResults
+    distPoint2Line (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2, const Eigen::Vector3f &p)
```

Comment:
There is a point to line distance function in `common` in "distances.h" can we reuse it?

Reply:
It looks like they are for checking against a line with infinite length, where I need to check against a line segment. Should I add these to this file? I would rename the methods to:

distPointToLineSeg
distLineSegToLineSeg

Reply:
Yes, please. I'd spell "segment" completely, and also move "distance" part to the end to be similar with already existing function names, i.e. `pointToLineSegmentDistance` and `lineSegmentToLineSegmentDistance`. What do you think?

Reply:
Sounds good.